reccmp/isledecomp/bin.py

```diff
@@ -25,38 +29,337 @@ class InvalidVirtualAddressError(IndexError):
+            part3 = struct.unpack_from(struct_fmt3, view, offset=offset)
+        else:
+            struct_fmt3 = "<III6H4I2H4I2I"
+            part3 = struct.unpack_from(struct_fmt3, view, offset=offset)
+        part3 = list(part3)
+        part3[13] = WindowsSubsystem(part3[13])
+        part3[14] = DllCharacteristics(part3[14])
+        offset += struct.calcsize(struct_fmt3)
+
+        count_directories = part3[-1]
+        directories = tuple(
+            DataDirectoryItemHeader(*item)
+            for item in struct.iter_unpack(
+                "<II", view[offset : offset + 8 * count_directories]
+            )
+        )
+        offset += 8 * count_directories
+        return cls(*part1, base_of_data, *part3, directories), offset
+
+
+class SectionFlags(IntFlag):
+    IMAGE_SCN_RESERVED_0X0 = 0x00000000
+    IMAGE_SCN_RESERVED_0X1 = 0x00000001
+    IMAGE_SCN_RESERVED_0X2 = 0x00000002
+    IMAGE_SCN_RESERVED_0X4 = 0x00000004
+    IMAGE_SCN_TYPE_NO_PAD = 0x00000008
+    IMAGE_SCN_RESERVED_0X10 = 0x00000010
+    IMAGE_SCN_CNT_CODE = 0x00000020
+    IMAGE_SCN_CNT_INITIALIZED_DATA = 0x00000040
+    IMAGE_SCN_CNT_UNINITIALIZED_DATA = 0x00000080
+    IMAGE_SCN_LNK_OTHER = 0x00000100
+    IMAGE_SCN_LNK_INFO = 0x00000200
+    IMAGE_SCN_RESERVED_0X400 = 0x00000400
+    IMAGE_SCN_LNK_REMOVE = 0x00000800
+    IMAGE_SCN_LNK_COMDAT = 0x00001000
+    IMAGE_SCN_GPREL = 0x00008000
+    IMAGE_SCN_MEM_PURGEABLE = 0x00020000
+    IMAGE_SCN_MEM_16BIT = 0x00020000
+    IMAGE_SCN_MEM_LOCKED = 0x00040000
+    IMAGE_SCN_MEM_PRELOAD = 0x00080000
+    # IMAGE_SCN_ALIGN_1BYTES = 0x00100000
+    # IMAGE_SCN_ALIGN_2BYTES = 0x00200000
+    # IMAGE_SCN_ALIGN_4BYTES = 0x00300000
+    # IMAGE_SCN_ALIGN_8BYTES = 0x00400000
+    # IMAGE_SCN_ALIGN_16BYTES = 0x00500000
+    # IMAGE_SCN_ALIGN_32BYTES = 0x00600000
+    # IMAGE_SCN_ALIGN_64BYTES = 0x00700000
+    # IMAGE_SCN_ALIGN_128BYTES = 0x00800000
+    # IMAGE_SCN_ALIGN_256BYTES = 0x00900000
+    # IMAGE_SCN_ALIGN_512BYTES = 0x00A00000
+    # IMAGE_SCN_ALIGN_1024BYTES = 0x00B00000
+    # IMAGE_SCN_ALIGN_2048BYTES = 0x00C00000
+    # IMAGE_SCN_ALIGN_4096BYTES = 0x00D00000
+    # IMAGE_SCN_ALIGN_8192BYTES = 0x00E00000
+    IMAGE_SCN_LNK_NRELOC_OVFL = 0x01000000
+    IMAGE_SCN_MEM_DISCARDABLE = 0x02000000
+    IMAGE_SCN_MEM_NOT_CACHED = 0x04000000
+    IMAGE_SCN_MEM_NOT_PAGED = 0x08000000
+    IMAGE_SCN_MEM_SHARED = 0x10000000
+    IMAGE_SCN_MEM_EXECUTE = 0x20000000
+    IMAGE_SCN_MEM_READ = 0x40000000
+    IMAGE_SCN_MEM_WRITE = 0x80000000
+
+
+@dataclass(frozen=True)
+class ImageSectionHeader:
+    name: str  # Name
+    virtual_size: int  # VirtualSize
+    virtual_address: int  # VirtualAddress
+    size_of_raw_data: int  # SizeOfRawData
+    pointer_to_raw_data: int  # PointerToRawData
+    pointer_to_relocations: int  # PointerToRelocations
+    pointer_to_line_numbers: int  # NumberOfRelocations
+    number_of_relocations: int  # NumberOfLineNumbers
+    number_of_line_numbers: SectionFlags  # Characteristics
```

Comment:
Should `characteristics` have its own member? I don't know if `number_or_relocations` or `number_of_line_numbers` are used for anything, but if we read both then the struct format should be `<8s6I2HI`

Reply:
Probably don't need the comments after each member either